client/app/components/alerts/AlertSubscriptions.jsx

```diff
@@ -0,0 +1,154 @@
+
+        destinations = destinations.filter(d => !includes(subscribedDestinations, d.id));
+
+        if (!includes(subscribedUsers, this.props.currentUser.id)) {
+          destinations.unshift({
+            name: this.props.currentUser.name + ' (Email)',
+            id: null,
+            icon: 'fa-envelope',
+          });
+        }
+        this.setState({
+          subscribers,
+          destinations,
+        });
+      });
+  }
+
+  state = {
+    selection: null,
+    subscribers: [],
+    destinations: [],
+  }
+
+  saveSubscriber = () => {
+    const sub = new this.props.AlertSubscription({ alert_id: this.props.alertId });
+    if (this.state.selection) {
+      sub.destination_id = this.state.selection;
+    }
+
+    sub.$save(
+      () => {
+        this.props.toastr.success('Subscribed.');
+        const destinations = this.state.destinations.filter(d => d.id !== this.state.selection);
+        this.setState({
+          subscribers: this.state.subscribers.concat(sub),
+          destinations,
+          selection: destinations.length ? destinations[0].id : null,
+        });
+      },
+      () => {
+        this.props.toastr.error('Failed saving subscription.');
+      },
+    );
+  }
+
+  unsubscribe = (subscriber) => {
+    const destination = subscriber.destination;
+    const user = subscriber.user;
+
+    subscriber.$delete(
+      () => {
+        this.props.toastr.success('Unsubscribed');
+        let destinations;
+        const subscribers = without(this.state.subscribers, subscriber);
+        if (destination) {
+          destinations = this.state.destinations.concat(destination);
+        } else if (user.id === this.props.currentUser.id) {
+          destinations = this.state.destinations.concat({
+            id: null,
+            name: this.props.currentUser.name + ' (Email)',
+            icon: 'fa-envelope',
+          });
+        }
+        const newState = { subscribers, destinations };
+        if (destinations.length === 1) {
+          newState.selection = destinations[0].id;
+        }
+        this.setState(newState);
+      },
+      () => {
+        this.props.toastr.error('Failed unsubscribing.');
+      },
+    );
+  }
+
+  updateSelection = selection => this.setState({ selection })
+
+  render() {
+    return (
+      <div className="p-5">
+        <h4>Notifications</h4>
+        <div>
+          <Select
+            value={this.state.selection || null}
+            disabled={this.state.destinations.length === 0}
+            onChange={this.updateSelection}
+          >
+            {this.state.destinations.map(d => <Select.Option value={d.id} key={d.name}><Destination destination={d} /></Select.Option>)}
+          </Select>
+        </div>
+        <div className="m-t-5">
+          <button
+            className="btn btn-default"
+            onClick={this.saveSubscriber}
+            disabled={this.state.destinations.length === 0}
+            style={{ width: '50%' }}
```

Comment:
Nit: this is probably inline since it was that way in angular but I still advise against it.